### PR TITLE
Stop rendering metrics for jobs that don't have them enabled

### PR DESCRIPTION
**Pull request by jbartok**
Fixes https://github.com/hazelcast/hazelcast-jet/issues/1569
Fixes https://github.com/hazelcast/hazelcast-jet/issues/1580

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/CompleteExecutionOperation.java [hunk 1]
```diff
@@ -44,14 +44,16 @@
 public class CompleteExecutionOperation extends Operation implements IdentifiedDataSerializable {
 
     private long executionId;
+    private boolean jobMetricsEnabled;
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/CompleteExecutionOperation.java]: shouldn't it be `saveMetrics` ?

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java [hunk 2]
```diff
@@ -63,6 +63,25 @@ public void setup() {
...
     }
 
+    @Test
+    public void when_jobMetricsDisabled_then_emptyMetrics() throws Throwable {
+        DAG dag = new DAG();
+        dag.newVertex("v1", MockP::new);
+        dag.newVertex("v2", (SupplierEx<Processor>) NoOutputSourceP::new);
+        Job job = jet().newJob(dag);
+
+        //when
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java]: The `// When` comment should be at the place where you do the _when_ action from the method name. Here it's where you disable the job metrics.

Also it's not clear to me whether you disable metrics collection or metrics saving. Looks like collection, but this PR is about metrics saving. Neither is disabled explicitly in this test.

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java [hunk 3]
```diff
@@ -63,6 +63,25 @@ public void setup() {
...
+        NoOutputSourceP.executionStarted.await();
+        assertJobStatusEventually(job, JobStatus.RUNNING);
+        //then
+        assertTrueAllTheTime(() -> assertEmptyJobMetrics(job), 1);
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java]: 1 second is pretty short, collection interval is 5 seconds by default, I don't see it shortened in this test.

**Comment by jbartok** [comment]
My understanding was lacking about the whole point of this issue. Will push improved changes in a sec.

**Comment by cangencer** [comment]
Please add tags + milestone

**Comment by jbartok** [comment]
Fixes https://github.com/hazelcast/hazelcast-jet/issues/1580

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/Job.java [hunk 4]
```diff
@@ -88,29 +88,26 @@ default String getName() {
...
+     * and {@link JobConfig#isMetricsEnabled() per job metrics collection}
+     * are enabled. Otherwise empty metrics will be returned.
      * <p>
-     * Once a job stops executing (successfully, after a failure, cancellation,
-     * or temporarily while suspended) the metrics will have their most
-     * recent values (i.e. the last metric values from the moment before the
-     * job completed).
+     * Keep in mind that the collections may occur at different times on
+     * each member, metrics from various members aren't from the same instant
+     * of time.
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/Job.java]: "instant of time" -> "instant"

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java [hunk 5]
```diff
@@ -451,7 +451,14 @@ public boolean isMetricsEnabled() {
 
     /**
      * Returns whether metrics should be stored in the cluster after the job
-     * completes. If enabled, metrics can be retrieved by calling {@link Job#getMetrics()}.
+     * completes. If enabled, metrics can be retrieved for all non-running
+     * jobs (have completed successfully, have failed, have been cancelled
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java]: 1. We normally document the set-methods and the get-methods refer to those. The reason is that the user mostly uses only the set-methods, the get-methods are used by Jet internally.

2. it doesn't apply to "all non-running jobs" but only to this job, when it's not running.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/CompleteExecutionOperation.java [hunk 6]
```diff
@@ -44,14 +44,16 @@
...
     private long executionId;
+    private boolean saveMetrics;
     private Throwable error;
     private RawJobMetrics response;
 
     public CompleteExecutionOperation() {
     }
 
-    public CompleteExecutionOperation(long executionId, Throwable error) {
+    public CompleteExecutionOperation(long executionId, boolean saveMetrics, Throwable error) {
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/operation/CompleteExecutionOperation.java]: I'd rename `saveMetrics` to `collectMetrics`. Metrics are not saved here.

**Pull request merged**